test/test_c10d.py

```diff
@@ -1640,13 +1641,18 @@ def tearDown(self):
     def world_size(self):
         return 2
 
+    @property
+    def delay_allreduce(self):
```

Comment:
There is `torch.distributed.all_reduce` instead of `allreduce` for the term "all-reduce". I suggest renaming to `delay_all_reduce` for consistency with already existing APIs.